src/app/components/blog/blog.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { BlogService } from '../../services/blog.service';
+import { Observable, catchError, of } from 'rxjs';
+import { BlogPost } from '../../models/Blog';
+
+@Component({
+  selector: 'app-blog',
+  templateUrl: './blog.component.html',
+  styleUrls: ['./blog.component.css'],
+})
+export class BlogComponent implements OnInit {
+  blogPosts$: Observable<BlogPost[]> | undefined;
+  loading: boolean = true;
+  error: string | undefined;
+
+  constructor(private blogService: BlogService) {}
+
+  ngOnInit(): void {
+    this.blogPosts$ = this.blogService.getPosts().pipe(
+      catchError((err) => {
+        this.error = 'Error loading posts. Please try again later.';
+        console.error(err);
+        return of([]);
+      })
+    );
+
+    this.blogPosts$.subscribe(() => {
+      this.loading = false;
```

Comment:
In the case of an error, the loading process will continue. It doesn't make sense to subscribe here just to set loading to false. Since signals are a new generation feature in Angular, consider using a signal instead